File: src/main/java/com/bot/commands/voice/DefaultVolumeCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.db.GuildDAO;
import com.jagrosh.jdautilities.command.CommandEvent;

public class DefaultVolumeCommand extends VoiceCommand {

    private GuildDAO guildDAO;

    public DefaultVolumeCommand() {
        this.name = "dvolume";
        this.arguments = "<Volume 1-200>";
        this.help = "Sets the default volume for the server";
        this.guildDAO = GuildDAO.getInstance();
    }

    @Override
    protected void executeCommand(CommandEvent commandEvent) {
        int newVolume;
        try{
            newVolume = Integer.parseInt(commandEvent.getArgs().split(" ")[0]);
            if (newVolume > 200 || newVolume < 0) {
                throw new NumberFormatException();
            }

            if (!guildDAO.updateGuildVolume(commandEvent.getGuild().getId(), newVolume)) {
                commandEvent.reply(commandEvent.getClient().getError() + " Something went wrong updating the default volume.");
                metricsManager.markCommandFailed(this, commandEvent.getAuthor(), commandEvent.getGuild());
                return;
            }

            commandEvent.reactSuccess();
        }
        catch (NumberFormatException e) {
            commandEvent.reply(commandEvent.getClient().getWarning() + " You must enter a volume between 0 and 200");
        }
    }

}

File: src/main/java/com/bot/commands/voice/DefaultVolumeCommand.kt
package com.bot.commands.voice

import com.bot.commands.ModerationCommand
import com.bot.db.GuildDAO
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class DefaultVolumeCommand : ModerationCommand() {
    private val guildDAO: GuildDAO

    init {
        name = "dvolume"
        arguments = "<Volume 1-200>"
        help = "Sets the default volume for the server"
        guildDAO = GuildDAO.getInstance()
    }

    @Trace(operationName = "executeCommand", resourceName = "dvolume")
    override fun executeCommand(commandEvent: CommandEvent) {
        val newVolume: Int
        try {
            newVolume = commandEvent.args.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0].toInt()
            if (newVolume > 200 || newVolume < 0) {
                throw NumberFormatException()
            }
            if (!guildDAO.updateGuildVolume(commandEvent.guild.id, newVolume)) {
                commandEvent.reply(commandEvent.client.error + " Something went wrong updating the default volume.")
                metricsManager.markCommandFailed(this, commandEvent.author, commandEvent.guild)
                return
            }
            commandEvent.reactSuccess()
        } catch (e: NumberFormatException) {
            commandEvent.reply(commandEvent.client.warning + " You must enter a volume between 0 and 200")
        }
    }
}